File: leetcode/9.回文数.cpp
/*
 * @lc app=leetcode.cn id=9 lang=cpp
 *
 * [9] 回文数
 */

// @lc code=start
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0){
            return false;
        }
        else if (x == 0) {
            return true;
        }
        else{
            int len;
            int num[99] = {0};
            for (len = 0;x ; ++len){
                num[len] = x % 10;
                x /= 10;
            };
            for (int i = 0; i <= len / 2; ++i){
                if (num[i] != num[len - i - 1]){
                    return false;
                }
            }
            return true;
        }
    }
};
// @lc code=end

File: leetcode/9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x = str(x)
        flag = 1
        for i in range(len(x) // 2):
            if x[i] != x[-i - 1]:
                flag = 0
                break
        if flag:
            return True
        else:
            return False
# @lc code=end

